pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -2495,8 +2495,23 @@ func GetUidFromUser(id string) string {
 	}
 	// split instances where the id may contain uid:gid
 	if strings.Contains(id, ":") {
-		return strings.Split(id, ":")[0]
+		id = strings.Split(id, ":")[0]
+	}
+	switch id {
+	case "root":
+		return "0"
+	case "nobody":
+		// On many os distros, nobody is default to 65534. However,
+		// **NOTICE** that this is not always true, for example on
+		// Fedora nobody is default to 99.
+		// This is fine for now, because we only need to know whether
+		// the user is root or not. However, in the future, this may
+		// become a problem if we rely on accurate uid.
+		return "65534"
```

Comment:
That's not right though. The UID we need is the mapping from the container's rootfs, not from the host distro. e.g. nobody can and will have different uids on the host OS and the container image, and in this case the string "nobody" is meant to be resolved by the runtime using the image's information.